### DISCUSSION TITLE
Build and Install

### DISCUSSION THREAD

**Discussion by cyclotron3k** [cardillan/mindcode#38] [General]
Related to Issue #27, and PR #35

I realise that the PR I made may have an issue, but I don't have any experience with Maven & Java projects so I wanted to throw it open for discussion before I attempt a solution.

In the updates to the documentation I submitted, under the "Native Installation" heading, I suggest the following:

> Set up the following environment variables:
> ```sh
> export SPRING_DATASOURCE_URL=jdbc:postgresql://localhost/mindcode_development
> export SPRING_DATASOURCE_USERNAME=postgres_username
> export SPRING_DATASOURCE_PASSWORD=postgres_password
> ```
> 
> Then run:
> 
> ```sh
> ./mvnw install
> ```

These environment variables override the database connection parameters that are defined in the two `application.properties` files, which is an issue because the `./mvnw install` command seems to run the unit tests. So now the unit tests and the web application will try to use the same database.

**Comment by francois** [comment]
Maven has a [`-DskipTests=true` option](https://maven.apache.org/plugins-archives/maven-surefire-plugin-2.12.4/examples/skipping-test.html). I think that would be the easiest way to prevent problems? A simple change to `bin/run-local` in #28 would be quick and painless to add. The README is already updated to use `bin/run-local`. What do you think?
